### DISCUSSION TITLE
Got 404 when reload the page in dynamic route in Gatsby on production only

### DISCUSSION THREAD

**Discussion by mic-tientpn** [gatsbyjs/gatsby#31407] [Help]
My app to deploy to heroku, i got 404 error when reload or direct access to client-only page

There are two situation:

1: Access home page (index.html in pages folder) -> access other client-only route page -> not get error 
2: Direct access to client-only route page -> get 404 error (But it still work)

# gatsby-node.js

```
exports.onCreatePage = ({ page, actions }) => {
  const { createPage } = actions
  if (page.path === `/`) {
    page.matchPath = `/*`
    createPage(page)
  }
}
```

I defined client-only route in file index.js

Any suggestion to resolve it?

Thanks!

**Comment by KyleAMathews** [comment]
See https://www.gatsbyjs.com/docs/how-to/routing/client-only-routes-and-user-authentication/#how-to-configure-your-hosting-service-to-handle-client-only-routes
